src/test/HostActivationTests/NativeHosting/HostContext.cs

```diff
@@ -224,166 +160,368 @@ public void GetDelegate_Multiple(string checkProperties)
+            {
+                SharedTestState.AppPropertyName,
+                SharedTestState.ConfigPropertyName
+            };
+            CommandResult result = Command.Create(sharedState.NativeHostPath, args.Concat(appArgs))
                 .CaptureStdErr()
                 .CaptureStdOut()
                 .EnvironmentVariable("COREHOST_TRACE", "1")
                 .EnvironmentVariable("DOTNET_ROOT", sharedState.DotNetRoot)
                 .EnvironmentVariable("DOTNET_ROOT(x86)", sharedState.DotNetRoot)
                 .EnvironmentVariable("TEST_BLOCK_MOCK_EXECUTE_ASSEMBLY", $"{sharedState.AppPath}.block")
+                .EnvironmentVariable("TEST_SIGNAL_MOCK_EXECUTE_ASSEMBLY", $"{sharedState.AppPath}.signal")
                 .Execute();
 
             result.Should().Pass()
-                .And.InitializeContextForApp(sharedState.AppPath)
-                .And.ExecuteAssemblyMock(sharedState.AppPath)
-                .And.InitializeSecondaryContext(sharedState.RuntimeConfigPath)
+                .And.ExecuteAssemblyMock(sharedState.AppPath, appArgs)
+                .And.InitializeSecondaryContext(sharedState.RuntimeConfigPath, CoreHostDifferentProperties)
                 .And.CreateDelegateMock();
 
-            switch(checkProperties)
+            CheckPropertiesValidation propertyValidation = new CheckPropertiesValidation(checkProperties, LogPrefix.App, SharedTestState.AppPropertyName, SharedTestState.AppPropertyValue);
+            if (scenario == Scenario.Mixed)
             {
-                case CheckProperties.None:
-                    result.Should()
-                        .HavePropertyMock(sharedState.AppPropertyName, sharedState.AppPropertyValue);
-                    break;
-                case CheckProperties.Get:
+                result.Should().InitializeContextForApp(sharedState.AppPath);
+                propertyValidation.ValidateActiveContext(result, SharedTestState.ConfigPropertyName);
+            }
+
+            propertyValidation.ValidateSecondaryContext(result, SharedTestState.ConfigPropertyName, SharedTestState.ConfigPropertyValue);
+        }
+
+        [Theory]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "1.1.0", Constants.RollForwardSetting.LatestMinor, false)]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "1.1.0", Constants.RollForwardSetting.LatestMajor, true)]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "2.1.0", Constants.RollForwardSetting.LatestPatch, false)]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "2.1.0", Constants.RollForwardSetting.LatestMinor, true)]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "2.1.0", Constants.RollForwardSetting.LatestMajor, true)]
+        [InlineData(Scenario.ConfigMultiple,  Constants.MicrosoftNETCoreApp, "3.1.0", Constants.RollForwardSetting.LatestMinor, false)]
```

Comment:
Can you please add at least for one of the scenarios also the other roll-forward values. I know that components should use the LatestMinor or LatestMajor most of the time, but the others should still work.